pgjdbc/src/main/java/org/postgresql/ds/common/BaseDataSource.java

```diff
@@ -758,6 +759,22 @@ public void setSslRootCert(@Nullable String file) {
     PGProperty.SSL_ROOT_CERT.set(properties, file);
   }
 
+  /**
+   * @param sslNegotiation one of SSLNegotiation.POSTGRES or SSLNegotiation.DIRECT
+   * @see PGProperty#SSL_NEGOTIATION
+   */
+  public void setSslNegotiation(@Nullable SSLNegotiation sslNegotiation) {
```

Comment:
Is there a reason to accept `null` values?

```suggestion
  public void setSslNegotiation(SSLNegotiation sslNegotiation) {
```

Reply:
However, we treat `null` as "remove the current value", so having `@Nullable` here makes sense.